Lib/Protocols/IdHL7.pas

```diff
@@ -184,25 +186,30 @@   EHL7CommunicationError = class(EIdException)
   NULL_IP = '0.0.0.0';  {do not localize}
   DEFAULT_CONN_LIMIT = 1;
   DEFAULT_RECONNECT_DELAY = 15000;
+  DEFAULT_CONNECTION_TIMEOUT = 0;
   DEFAULT_COMM_MODE = cmUnknown;
   DEFAULT_IS_LISTENER = True;
   MILLISECOND_LENGTH = (1 / (24 * 60 * 60 * 1000));
+  SEND_RESPONSE_NAMES : Array [TSendResponse] Of String = ('None', 'Error', 'NoConnection', 'Sent', 'OK', 'Timeout');
 
 type
   // the connection is provided in these events so that applications can obtain information about the
   // the peer. It's never OK to write to these connections
-  TMessageArriveEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String) of object;
-  TMessageReceiveEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String; var VHandled: Boolean; var VReply: String) of object;
-  TReceiveErrorEvent = procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: String; AException: Exception; var VReply: String; var VDropConnection: Boolean) of object;
+  TMessageArriveEvent = Procedure(ASender: TObject; AConnection: TIdTCPConnection; AMsg: AnsiString) Of Object;

```

Comment:
I'm not a fan of forcing `AnsiString` everywhere.  The rest of Indy uses the native `String` type, converting to/from bytes using the `IIdTextEncoding` interface as needed.  Most Internet protocols are 7bit/8bit data, and Indy handles them just fine, even in Unicode environments.  I would prefer that this unit does the same, for consistency with the rest of the Indy library.  Also, because `AnsiString` is charset-aware in D2009+, but this code is not handling charsets at all.  If specific charsets are needed, consider introducing `IIdTextEncoding` properties/parameters as needed to let the caller decide which charsets to use when conversions are needed.

Reply:
Made changes regarding this, introduced DefSendEncoding now to address this (tested works nice)
https://github.com/IndySockets/Indy/pull/442/commits/7bd8d29be4574608893391b97fcb794d959042b9